apps/payment/models.py

```diff
@@ -553,3 +580,106 @@ def _send_receipt(self, payment_date, subject, description, payment_id, items, t
 
     class Meta:
         default_permissions = ('add', 'change', 'delete')
+
+
+class FikenSale(models.Model):
+    stripe_key = models.CharField(
+        'Stripe key',
+        max_length=20,
+        choices=StripeKey.ALL_CHOICES,
+        default=StripeKey.ARRKOM,
+    )
+    transaction_type = models.CharField(
+        'Transaction Type',
+        max_length=20,
+        choices=TransactionType.ALL_CHOICES,
+    )
+    original_amount = models.FloatField()
```

Comment:
Don't use floats for money. Especially if stored internally and is later used for calculations^1.

Here's a nice, short read: https://husobee.github.io/money/float/2016/09/23/never-use-floats-for-currency.html

I'd suggest to keep the "Stripe way", as in keeping the base "øre" amount ("100.50 kr" would be "10050 øre") and use that. That way you never have to deal with floating point arithmetic and its problems. If Fiken accepts only "kroner and øre" as a float (first off: WTF), convert it when posting to their API rather than storing this internally.

Why?
```python
>>> 0.1 + 0.2
0.30000000000000004
```
Source: https://docs.python.org/3.7/tutorial/floatingpoint.html

1) It might not be a problem here because it is only stored in this class to later be posted to Fiken, but better handle it correctly from the first time around IMO. Suddenly this is getting rewritten without people thinking about this and then you get the problems.

---

So I'd suggest creating a "to_fiken_api_format" function on the class which converts the model to a JSON representation that Fiken accepts, which converts the model class into whatever format Fiken expects.

Reply:
Sure, I can store it as an integer. The problem is that I still have to calculate the stripe fee and vat which will lead to floating point numbers?